packages/soci-snapshotter/soci-snapshotter.spec

```diff
@@ -42,6 +42,24 @@ Conflicts: (%{_cross_os}image-feature(no-fips) or %{name}-bin)
 %description fips-bin
 %{summary}.
 
+%package cli-bin
```

Comment:
You should provide a package `cli`, that requires `%{name}-cli(binaries)` so that the downstreams only require `%{cross_os}-soci-cli` and through RPM magic, they get the correct version (FIPS or No FIPS).

You will have to adjust the subpackages dependencies and conflicts accordingly.